src/Controller/Api/UserController.ts

```diff
@@ -36,22 +36,16 @@ export class UserController {
     public async loginUser(req: ValidatedRequest<UserRequestSchema>, res: Response) {
         try {
             const {email, password} = req.body;
-            const user = await this.userService.getUserByEmail(email);
-            if (user) {
-                const isValidPassword = bcrypt.compareSync(password, user['hash_password']);
-                if (!isValidPassword) {
-                    res.status(401).json({error: "Unauthorized user."});
-                } else {
-                    const token = jwt.sign(user, config.jwtKey, {expiresIn: 86400});
-                    res.status(200).json({token: token})
-                }
+            const authenticatedUser = await this.userService.isAuthenticatedUser(email, password);
+            if (authenticatedUser) {
```

Comment:
I more had in mind that you move all of this if/else logic (that contains jwt dependency) in the service i.e. function `authenticateUser` and here just catch an error.

However, you will have multiple error types in that case and to generalize them, you can create new custom Error class that contains properties `status` and `message`. 

```
errors.ts
export class ServiceError extends Error {
   public status;

  constructor(status, message?: string) {
    super(message);

    this.status = status;
    this.name = "FailedServiceAction";
    this.stack = new Error().stack;
  }
}

and you can then just catch `e` it here and pass to res.status(e.status).json({ error: e.message });
```

The reason why we need a new class for this is to extend existing Error prototype with our properties (types). I hope that makes sense.